geoportal_1/src/main/java/org/opengeoportal/ogc/wcs/wcs1_0_0/WcsGetCoverage1_0_0.java

```diff
@@ -46,53 +46,73 @@ public static String createWcsGetCoverageRequest(String layerName, CoverageOffer
 			Hints hints = new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, Boolean.TRUE);
 			CRSAuthorityFactory factory = ReferencingFactoryFinder.getCRSAuthorityFactory("EPSG", hints);
 			CoordinateReferenceSystem sourceCRS = factory.createCoordinateReferenceSystem("EPSG:4326");
-			
+		    //The bounding box is created in EPSG 4326.  The line below takes the corrdinates given and references to the sourceCRS (EPSG:4326).
+		    //The envelope is then transformed to the native (targetCRS) bounds below.		    
 		    ReferencedEnvelope envelope = new ReferencedEnvelope(bounds.getMinX(), bounds.getMaxX(), bounds.getMinY(), bounds.getMaxY(), sourceCRS);
 		    logger.info("requested Bounds: " + OgpUtils.referencedEnvelopeToString(envelope));
+		    
 		    // Transform using 10 sample points around the envelope
-
 		    CoordinateReferenceSystem targetCRS = factory.createCoordinateReferenceSystem(coverageOffering.getSupportedCRSs().get(0));
 
 		    ReferencedEnvelope result = envelope.transform(targetCRS, true, 10);
 
-		 String getCoverageRequest = "service=WCS&version=" + VERSION + 
-		 		"&request=GetCoverage&coverage=" + layerName +
-				"&bbox=" + OgpUtils.referencedEnvelopeToString(result) +
-				"&crs=" + coverageOffering.getSupportedCRSs().get(0) + 
-				"&format=" + outputFormat;
-		 
-		getCoverageRequest += generateSize(coverageOffering.getRectifiedGrid());
-		
-    	return getCoverageRequest;
-	}
-	
-	private static String generateSize(RectifiedGrid rgrid) throws Exception{
-		Double resx = rgrid.getResx();
-		Double resy = rgrid.getResy();
+		// Pull min and max values from full coverage envelope
+		ReferencedEnvelope nativeRefEnv = new ReferencedEnvelope(nativeBbox.getMinX(), nativeBbox.getMaxX(), nativeBbox.getMinY(), nativeBbox.getMaxY(), targetCRS);
 		
-		String size = "&";
+		Double resX = coverageOffering.getRectifiedGrid().getResx();  //Get X Resolution of Raster
+		Double resY = coverageOffering.getRectifiedGrid().getResy();  //Get Y Resolution of Raster
+
+		Double xMin, yMin, xMax, yMax;
+
+		// Sending a request for a bounding box which cut through pixels caused geotools to warp the coverage compromising the integrity and altering noData values
+		// This was an an issue 8 bit unsigned 4 band rasters which were having noData values set to 0 causing noData values to be read where r,g,b,i were something like 50,0,190,89
+		if ( nativeRefEnv.getMinimum(0) > result.getMinimum(0)){  //Checks if xMin is less than xMin of whole coverage.  If so, set xMin to xmin of whole coverage 
```

Comment:
The following section is where the bounding box is snapped to whole pixels rather than arbitrarily cutting through them.